react-client-app/src/components/DatasetTbl.jsx

```diff
@@ -0,0 +1,138 @@
+import { Flex, Spinner, Table, Thead, Tbody, Tr, Th, Td, Spacer, TableCaption, Link, Button, HStack, Select, Text } from "@chakra-ui/react"
+import { useGet, usePatch, chunkIds } from '../api/values'
+import PageHeading from './PageHeading'
+import ConceptTag from './ConceptTag'
+import moment from 'moment';
+
+const DatasetTbl = (props) => {
+    const [datasets, setDatasets] = useState({})
+    const [loadingMessage, setLoadingMessage] = useState("Loading Datasets")
+    const [datapartnerFilter, setDataPartnerFilter] = useState("All");
+    const [title, setTitle] = useState("Datasets");
+
+    useEffect(async () => {
+        // run on initial page load
+        props.setTitle(null);
+        // get datasets and sort by id
+        let datasets = await useGet(`/datasets/`);
+        datasets = datasets.sort((b, a) => (a.id > b.id) ? 1 : ((b.id > a.id) ? -1 : 0));
+        setDatasets(datasets);
+        // for each dataset use the data partner and admin ids to get name to display
+        // get list of unique data partner and admin ids
+        const adminObject = {};
+        datasets.map((dataset) => {
+            adminObject[dataset.admin] = true;
+            const created_at = {};
+            created_at.created_at = dataset.created_at;
+            created_at.displayString = moment(dataset.created_at.toString()).format("MMM. DD, YYYY, h:mm a");
+            dataset.created_at = created_at;
+        });
+
+        const dataPartnerObject = {};
+        datasets.map((dataset) => {
+            dataPartnerObject[dataset.data_partner] = true;
+        });
+        const dataPartnerIds = chunkIds(Object.keys(dataPartnerObject));
+        const dataPartnerPromises = [];
+        for (let i = 0; i < dataPartnerIds.length; i++) {
+            dataPartnerPromises.push(useGet(`/datapartners/?id__in=${dataPartnerIds[i].join()}`));
+        }
+        let dataPartners = await Promise.all(dataPartnerPromises)
+        dataPartners = dataPartners[0]
+        dataPartners.forEach((element) => {
+            datasets = datasets.map((dataset) => dataset.data_partner == element.id ? { ...dataset, data_partner: element } : dataset);
+        });
+        setDatasets(datasets);
+        setLoadingMessage(null);
+
+    }, []);
+
+    const applyFilters = (variable) => {
+        let newData = variable.map(dataset => dataset)
+
+        if (datapartnerFilter !== "All") {
+            newData = newData.filter(dataset => dataset.data_partner.name === datapartnerFilter)
+        }
+        return newData
+    }
+
+    const removeFilter = (a, b) => {
+        if (a.includes("Data Partner")) {
+            setDataPartnerFilter("All")
+        }
+    }
+
+    if (loadingMessage) {
+        //Render Loading State
+        return (
+            <div>
+                <Flex padding="30px">
+                    <Spinner />
+                    <Flex marginLeft="10px">{loadingMessage}</Flex>
+                </Flex>
+            </div>
```

Comment:
We use this in tonnes of places. We should convert to a component in another PR

Reply:
Agreed, keep for another PR